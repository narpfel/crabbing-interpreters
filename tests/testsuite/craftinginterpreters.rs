use std::path::PathBuf;
use std::process::Command;

use insta_cmd::assert_cmd_snapshot;
use insta_cmd::get_cargo_bin;
use rstest::rstest;

use crate::as_relative;

#[rstest]
fn tests(
    #[files("craftinginterpreters/test/*.lox")]
    #[files("craftinginterpreters/test/assignment/**/*.lox")]
    #[files("craftinginterpreters/test/block/**/*.lox")]
    #[files("craftinginterpreters/test/bool/**/*.lox")]
    #[files("craftinginterpreters/test/comments/**/*.lox")]
    #[files("craftinginterpreters/test/if/**/*.lox")]
    #[files("craftinginterpreters/test/logical_operator/**/*.lox")]
    #[files("craftinginterpreters/test/nil/**/*.lox")]
    #[files("craftinginterpreters/test/number/**/*.lox")]
    #[files("craftinginterpreters/test/operator/**/*.lox")]
    #[files("craftinginterpreters/test/print/**/*.lox")]
    #[files("craftinginterpreters/test/string/**/*.lox")]
    path: PathBuf,
) {
    let path = as_relative(&path);
    assert_cmd_snapshot!(
        path.display().to_string(),
        Command::new(get_cargo_bin("crabbing-interpreters")).arg(path)
    )
}
