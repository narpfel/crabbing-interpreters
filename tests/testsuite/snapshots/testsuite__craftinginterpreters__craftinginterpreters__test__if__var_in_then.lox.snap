---
source: tests/testsuite/craftinginterpreters.rs
info:
  program: crabbing-interpreters
  args:
    - craftinginterpreters/test/if/var_in_then.lox
---
success: false
exit_code: 65
----- stdout -----

----- stderr -----
[line 2] Error at 'var': Expect expression.
